dart/dynamics/SimpleFrame.cpp

```diff
@@ -120,6 +132,13 @@ SimpleFrame& SimpleFrame::operator=(const SimpleFrame& _otherFrame)
 }
 
 //==============================================================================
+std::shared_ptr<SimpleFrame> SimpleFrame::createChildSimpleFrame(
```

Comment:
I'm slightly concerned about this function, because I'm afraid it might imply that the child you create will be held by this ``SimpleFrame`` instance. In other words, the user might create a bunch of children for this SimpleFrame and assume that it's sufficient to hold onto a reference to just the parent SimpleFrame in order to keep all the created children alive. I think we should either:

1. Make it extremely clear in the documentation that the user is responsible for hanging onto the created SimpleFrame. We should also use a different word besides ``create``, because in all other cases where you call ``createXXX()`` on an object instance, that object instance will hang onto the created instance for you. Maybe change the name to ``spawnChildSimpleFrame`` just to make it clear that the behavior is different from other ``createXXX()`` functions.

2. Allow SimpleFrames to hold onto a ``std::vector<SimpleFramePtr>`` for child frames. This would allow a user to hold onto an entire tree of SimpleFrames just by holding onto a reference to the root ``SimpleFrame``. The drawback to this is a user might think that the SimpleFrame tree will behave like a Skeleton where holding onto any member of it is sufficient to hold onto all of them, but that would not be the case here.